server/test/repository/ApplicationRepositoryTest.java

```diff
@@ -216,14 +217,83 @@ public void getApplications() {
     assertThat(repo.getApplications(restrictiveFilter)).isEmpty();
   }
 
+  @Test
+  public void getApplicationsForApplicant() throws Exception {
+    Applicant applicant = saveApplicant("Applicant");
+
+    Program program = createProgram("Program");
+    Application appDraft1 = Application.create(applicant, program, LifecycleStage.DRAFT);
+    appDraft1.save();
+    Application appDraft2 = Application.create(applicant, program, LifecycleStage.DRAFT);
+    appDraft2.save();
+    Application appActive1 = Application.create(applicant, program, LifecycleStage.ACTIVE);
+    appActive1.save();
+    Application appActive2 = Application.create(applicant, program, LifecycleStage.ACTIVE);
+    appActive2.save();
+    Application appObsolete1 = Application.create(applicant, program, LifecycleStage.OBSOLETE);
+    appObsolete1.save();
+    Application appObsolete2 = Application.create(applicant, program, LifecycleStage.OBSOLETE);
+    appObsolete2.save();
+
+    ImmutableSet<Application> result =
+        repo.getApplicationsForApplicant(
+                applicant.id,
+                ImmutableSet.of(
+                    LifecycleStage.DRAFT, LifecycleStage.ACTIVE, LifecycleStage.OBSOLETE))
+            .toCompletableFuture()
+            .get();
+    assertThat(result.stream().map(a -> a.id))
+        .containsExactlyInAnyOrder(
+            appDraft1.id,
+            appDraft2.id,
+            appActive1.id,
+            appActive2.id,
+            appObsolete1.id,
+            appObsolete2.id);
+
+    result =
+        repo.getApplicationsForApplicant(applicant.id, ImmutableSet.of(LifecycleStage.DRAFT))
+            .toCompletableFuture()
+            .get();
+    assertThat(result.stream().map(a -> a.id))
+        .containsExactlyInAnyOrder(appDraft1.id, appDraft2.id);
+
+    result =
+        repo.getApplicationsForApplicant(applicant.id, ImmutableSet.of())
+            .toCompletableFuture()
+            .get();
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  public void getApplicationsForApplicant_filtersById() throws Exception {
+    Applicant primaryApplicant = saveApplicant("Applicant");
+    Applicant otherApplicant = saveApplicant("Other");
+
+    Program program = createProgram("Program");
+    Application appDraft1 = Application.create(primaryApplicant, program, LifecycleStage.DRAFT);
+    appDraft1.save();
+    Application appDraft2 = Application.create(primaryApplicant, program, LifecycleStage.DRAFT);
+    appDraft2.save();
+
+    ImmutableSet<Application> result =
+        repo.getApplicationsForApplicant(
+                otherApplicant.id,
+                ImmutableSet.of(
+                    LifecycleStage.DRAFT, LifecycleStage.ACTIVE, LifecycleStage.OBSOLETE))
+            .toCompletableFuture()
+            .get();
+    assertThat(result).isEmpty();
```

Comment:
Suggestion: make an application for both and ensure each applicant get the right application.

As is if the method always returned nothing this would pass.

Reply:
Done.